### PR TITLE
Get Tx Status

### PR DESCRIPTION
**Pull request by NateWilliams2**
- Handlers that submit transactions to the vochain (createOrganization, updateOrganization, createProcess, setProcessStatus) create a tx hash and add it to a map, along with the time of tx creation
- Handlers that submit txs and update the database store the database queries on a sync.Map
- GetTransactionStatus handler determines if a transaction has been "mined" (if it was created more than 15 seconds ago) and submits the pending database query if so

### CODE REVIEW

File: urlapi/txhash.go [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+
+// GET https://server/v1/priv/transactions/<transactionHash>
+func (u *URLAPI) getTxStatusHandler(msg *bearerstdapi.BearerStandardAPIdata,
+	ctx *httprouter.HTTPContext) error {
+	txHash, err := util.GetBytesID(ctx, "transactionHash")
+	if err != nil {
+		return err
+	}
+	txTime, ok := u.txWaitMap[hex.EncodeToString(txHash)]
+	mined := false
```
Comment (p4u) [inline urlapi/txhash.go:54]: `mined := ok && txTime.Add(15*time.Second).Before(time.Now())`

File: urlapi/txhash.go [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+	if ok && txTime.Add(15*time.Second).Before(time.Now()) {
+		mined = true
+	}
+	// TODO make vocclient api request to get tx status
+	// If tx has been mined, check dbTransactions map for pending db queries
+	if mined {
```
Comment (p4u) [inline urlapi/txhash.go:60]: reverse this logic, it makes code easier to read:

`if !mined { return ... }`

File: urlapi/urlapi.go [hunk 3]
```diff
@@ -29,6 +31,10 @@ type URLAPI struct {
 	metricsagent *metrics.Agent
 	db           database.Database
 	vocClient    *vocclient.Client
+	// Map of database queries pending transactions being mined
+	dbTransactions sync.Map
+	// TODO remove temporary tx time map
+	txWaitMap map[string]time.Time
```
Comment (p4u) [inline urlapi/urlapi.go:37]: This map seems racy

**Review by p4u — CHANGES_REQUESTED** [review]

File: types/api.go [hunk 4]
```diff
@@ -55,6 +55,7 @@ type APIResponse struct {
 	ProcessID        types.HexBytes       `json:"processId,omitempty"`
 	PublicProcesses  []APIElectionSummary `json:"public,omitempty"`
 	Registered       bool                 `json:"registered,omitempty"`
+	TxHash           string               `json:"txHash,omitempty"`
```
Comment (p4u) [inline types/api.go:58]: Wouldn't be better to use types.HexBytes here?

**Review by p4u — APPROVED** [review]

**Comment by p4u** [comment]
I think the typescript part might require @brickpop review, but up to you.

File: types/api.go (hunk 4 above)
Comment (NateWilliams2) [reply types/api.go:58]: The client would only ever use this to query for the tx status, in which case they would append the txHash string to the request URL. So from their perspective, a string is fine. Also, we use the txHash as a string internally in the tx maps on the API server, so we will have to convert the txHash to a string when we receive it from the vochain anyways. I'm fine changing it to HexBytes, but more type conversion would be required than the current case.

**Comment by emmdim** [comment]
> I think the typescript part might require @brickpop review, but up to you.

He can review in a later iteration. It works btw

**Pull request merged**